.github/workflows/codeql.yml

```diff
@@ -0,0 +1,42 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  schedule:
+    - cron: "1 18 * * 0"
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ cpp, python ]
```

Comment:
```suggestion
        language: [ c, python ]
```

Reply:
I assume this is a miss detection of cpp vs c. Not sure if it matters -- for the purposes of code scanning one is a subset of the other. -- I also don't know if this change break the action.

Reply:
`cpp` is the setting for both c and cpp checks, but there is a `c` alias that translates to `cpp`, so setting this to `c` is probably good: https://github.com/github/codeql-action/blob/006bb00765df5ace47e70657a677e6ecd3b0d6a1/src/languages.ts#L5-L15